core/drivers/crypto/se050/session.c

```diff
@@ -64,32 +68,62 @@ static TEE_Result enable_scp03(void)
 	return TEE_SUCCESS;
 }
 
-static TEE_Result se050_early_init(void)
+static TEE_Result se050_early_init_default(void)
 {
-	TEE_Result ret = TEE_SUCCESS;
-
-	ret = se050_core_early_init(NULL);
-	if (ret) {
+	if (se050_core_early_init(NULL)) {
 		EMSG("Failed to open the default session");
-		goto out;
+		panic();
 	}
 
-	ret = update_se_info();
-	if (ret) {
+	if (update_se_info()) {
 		EMSG("Failed to read the secure element configuration");
-		goto out;
+		panic();
 	}
 
 	if (IS_ENABLED(CFG_CORE_SE05X_SCP03_EARLY)) {
-		ret = enable_scp03();
-		if (ret)
+		if (enable_scp03()) {
 			EMSG("Failed to open the SCP03 session");
+			panic();
+		}
 	}
-out:
-	if (ret)
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result se050_early_init_scp03(void)
+{
+	/* Initialize session */
+	se050_session = (sss_se05x_session_t *)((void *)&se050_ctx.session);
+	se050_kstore = (sss_se05x_key_store_t *)((void *)&se050_ctx.ks);
+
+#ifdef CFG_CORE_SE05X_OEFID
```

Comment:
`if IS_ENABLED(CFG_CORE_SE05X_OEFID) {`

Reply:
um, does this apply to strings as well? ie, will I have to define a default?

Reply:
No, you're right. We would need a special macro for this. The `#if` is fine then.